### ISSUE TITLE
"RangeError: Maximum call stack size exceeded" in source code block titles with mutual cross references

### ISSUE THREAD

**Issue by leorochael** [asciidoctor/asciidoctor-vscode#438]
## Description

If a file contains source code blocks with titles and `[[references]]`, and the titles themselves `<<cross-reference>>` one another, the error in the screenshot below appears near the bottom-right corner of VS.Code.

When I try to find more information in the Output consoles, none of them have that error message in them. Not the `Exensions` output, nor any of the `Log(*)` consoles like `Log (Extension Host)`. The only output related to asciidoctor that I found was:

```
[2021-09-10 13:35:16.702] [exthost] [info] ExtensionService#_doActivateExtension asciidoctor.asciidoctor-vscode {"startup":false,"extensionId":{"value":"asciidoctor.asciidoctor-vscode","_lower":"asciidoctor.asciidoctor-vscode"},"activationEvent":"onLanguage:asciidoc"}
[2021-09-10 13:35:16.702] [exthost] [info] ExtensionService#loadCommonJSModule file:///.../.vscode/extensions/asciidoctor.asciidoctor-vscode-2.8.9/dist/src/extension.js
```

## System Information

AsciiDoc v2.8.9
VS.Code 1.60.0
OS: Linux x64 5.4.0-81-generic

## To Reproduce

1. Download and rename the attached file to remove `.txt` extension and leave the `.adoc` extension.
2. Type something and save.
3. See an alert like in the screenshot in the lower right region of the window.

## Screenshots & Files

Attached file that reproduces the error:

[mutual-reference-causes-rangeerror-maximum-call-stack-size-exceeded.adoc.txt](https://github.com/asciidoctor/asciidoctor-vscode/files/7145744/mutual-reference-causes-rangeerror-maximum-call-stack-size-exceeded.adoc.txt)

Screenshot of the error message in the bottom-right corner of the window:

![image](https://user-images.githubusercontent.com/239125/132889386-311e2e36-1a78-4222-8c28-492c5fbaa1f8.png)

## Additional Context

N/A

**Comment by leorochael** [comment]
How can I find out more information about the error?

Is there some additional logging that I can enable?

**Comment by leorochael** [comment]
The description previously included this text, that I removed now that I was able to locate a minimal example:

> In a longish (2555) files that I have, I'm getting the alert reported in the screenshot below in the bottom right corner of the VS.Code window.
>
> I'm trying to bissect the file to find what asciidoc source causing it, but I cannot share the whole file right now.

**Comment by Mogztter** [comment]
@joaompinto I don't recall how to perform a new release? Is the release process fully automated using GitHub Action? (i.e., using https://github.com/asciidoctor/asciidoctor-vscode/blob/master/.github/workflows/release.yml).

If so, the only thing I need to do is push a new tag on the main branch right?

I think we have enough bug fixes to cut a new release? Any objection?

**Comment by danyill** [comment]
> @joaompinto I don't recall how to perform a new release? Is the release process fully automated using GitHub Action? (i.e., using https://github.com/asciidoctor/asciidoctor-vscode/blob/master/.github/workflows/release.yml).
> 
> If so, the only thing I need to do is push a new tag on the main branch right?
> 
> I think we have enough bug fixes to cut a new release? Any objection?

@Mogztter FWIW that's correct. Just a CHANGELOG update required I guess. I'm keen for a release.

One day I hope to write some tests for this extension to function as a security blanket because the automatic updating of end-users makes me afraid. But at least the automation should allow problems to be resolved quickly...

**Comment by joaompinto** [comment]
@danyill , yes, it's fully automated, you just need to create a tag vx.y.z, it will build and publish.

**Comment by danyill** [comment]
This should be addressed in the latest release (`v2.8.10`). 
If this does not adequately address this issue, please feel free to re-open.
